core/base/channel.go

```diff
@@ -37,6 +43,29 @@ func isRisk(potential string) bool {
 	return false
 }
 
+// isMoreStableThan returns true if r is more stable than other.
+// Stability order: stable > candidate > beta > edge.
+func (r Risk) isMoreStableThan(other Risk) bool {
+	return r.index() < other.index()
+}
+
+// index returns the stability index of a risk.
+// Lower index means more stable.
+func (r Risk) index() int {
+	switch r {
+	case Stable:
+		return 0
+	case Candidate:
+		return 1
+	case Beta:
+		return 2
+	case Edge:
+		return 3
+	default:
+		return 4 // This should not happen, but unknown risks are least stable.
```

Comment:
todo: If we ever get this, it's bad data from charm hub which needs to be reported. It should fail.